### ISSUE TITLE
SC2005: exit code changes if you follow the suggestion, suggestion should note this

### ISSUE THREAD

**Issue by unhammer** [koalaman/shellcheck#2078]
(Not sure if this should be considered a bug or what.)

- Rule Id SC2005
- My shellcheck version "online" and 0.7.0
- [until my edit] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
set -e
echo "$(make)"

```

#### Here's what shellcheck currently says:
```
Line 3:
echo "$(make)"
     ^-- SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
```

#### Here's what I wanted or expected to see:

```
Line 3:
echo "$(make)"
     ^-- SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'. If using set -e or relying on exit code, use 'cmd || true'.
```

The command `echo "$(false)` will return true, and the script will keep going even when `set -e` or followed by `&&`, so it's not a suggestion that you can blindly apply.

**Comment by brother** [comment]
> ...so it's not a suggestion that you can blindly apply.

isn't that true for most of the suggestions? you have the option to ignore the warning as a false positive with a directive. Everything is "use with care, apply brain first".

**Comment by unhammer** [comment]
> Everything is "use with care, apply brain first".

Of course, but it doesn't cost shellcheck much to notify of the pitfalls of applying a suggestion.
